Use tags map for tracking server group resources

milismsft opened the pull request:
Use the tags map to track dependencies between various Azure resources.

Commit:
provider/docker: Paginage catalog results

gregturn committed:
Stop using cloudFoundryCloudProvider.id

Commit:
provider/kubernetes: Block until kubernetes state is consistent

Commit:
provider/kubernetes: Provide missing instances to kubernetes load balancers

Commit:
provider/kubernetes: Provide deck with operation to upsert given lb

lwander committed:
Merge pull request #459 from lwander/kubernetes-locking-wait

provider/kubernetes: Block until kubernetes state is consistent

cfieber committed:
Merge pull request #438 from gregturn/cf-id

Stop using cloudFoundryCloudProvider.id

milismsft committed:
Track various relationships between server group and its resources via the tag map. Bug fix: retrieving load balancers with the same name across different locations/regions

anotherchrisberry committed:
enhancements to scaling policy upsert operations

milismsft commented:
@JargoonPard @scotmoor PTAL

lwander committed:
Merge pull request #460 from lwander/kubernetes-load-balancer-report-instances

provider/kubernetes: Provide missing instances to kubernetes load balancers

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy:
```diff
@@ -36,12 +36,15 @@ class AzureServerGroupDescription extends AzureResourceOpsDescription {
+    long min
+    long max
+    Boolean useSourceCapacity
   }
```
What is the purpose of the AzureScaleSetCapacity? There is no mapping for the min/max fields onto a scale set.

milismsft replied:
These values get passed in as part of the deployment of a server group in a pipeline (see Capacity tab). There's no actual usage of these at the moment at least not until we support autoscaling (in AWS these values are used when creating an autoscaling ASG).

JargoonPard replied:
Sure, but since we are not using them please remove them until we add our own autoscaling support as it will likely look quite different.

lwander committed:
Merge pull request #431 from lwander/docker-paginate-catalog

provider/docker: Paginage catalog results

cfieber committed:
replace direct construction of NetflixAmazonCredentials with TestCredential support class.

NetflixAmazonCredentials has many attributes, and adding to those attributes ripples through the test codebase

cfieber committed:
adds type info to ProviderUtils.buildThreadSafeSetOfAccounts

cfieber committed:
Adds an includePublicImages flag to AWS ImageCachingAgent. Public images will
only be cached by the agent if that flag is set.

When configuring ImageCachingAgents, only set includePublicImages once per region.

cfieber committed:
Merge pull request #464 from cfieber/reduce_aws_image_caching

Reduce aws image caching

milismsft commented:
@JargoonPard @scotmoor during further testing I found that the tags did not properly capture the storage account name (suffix was missing). I have that fixed now.

I also added a fix for the load balancer viewing bug (literally two lines change).

Commit:
provider/kubernetes: Delete security group

lwander committed:
Merge pull request #466 from lwander/kubernetes-security-group-delete

provider/kubernetes: Delete security group

lwander committed:
Merge pull request #462 from lwander/kubernetes-load-balancer-description

provider/kubernetes: Load balancer description attribute

Commit:
provider/kubernetes: Prepare for deploying spinnaker on spinnaker

anotherchrisberry committed:
Merge pull request #463 from anotherchrisberry/scaling-policy-cud

enhancements to scaling policy upsert operations

Commit:
provider/kubernetes: Report load balancer <-> security group

Commit:
Merge pull request #467 from lwander/kubernetes-assorted-container-specs

provider/kubernetes: Prepare for deploying spinnaker with spinnaker

Commit:
Merge pull request #470 from lwander/kubernetes-attach-sgs-to-lbs

provider/kubernetes: Report load balancer <-> security group
